controls/cusp_fedora.yml

```diff
@@ -0,0 +1,517 @@
+      - accounts_tmout
+      - var_accounts_tmout=15_min
+      - accounts_root_gid_zero
+      - accounts_umask_etc_bashrc
+      - accounts_umask_etc_login_defs
+      - accounts_umask_etc_profile
+      - var_accounts_user_umask=027
+      - account_password_selinux_faillock_dir
+      - enable_authselect
+      #  password requirements
+      - accounts_password_pam_pwquality_password_auth
+      - accounts_password_pam_pwquality_system_auth
+      - accounts_password_pam_maxrepeat
+      - accounts_password_pam_minclass
+      - accounts_password_pam_minlen
+      - accounts_password_pam_retry
+      - var_password_pam_minclass=3
+      - var_password_pam_minlen=15
+      #  password reuse
+      - var_password_pam_remember_control_flag=requisite_or_required
+      - var_password_pam_remember=5
+      - accounts_password_pam_difok
+      - var_password_pam_difok=8
+
+  - id: 4.2
+    title: Sudo
+    description: >-
+      secure sudo configuration
+    status: automated
+    rules:
+      - package_sudo_installed
+      - sudo_add_use_pty
+      - sudo_custom_logfile
+      - sudo_require_authentication
+      - sudo_require_reauthentication
+      - use_pam_wheel_for_su
+      - sudoers_default_includedir
+
+  - id: 4.3
+    title: SSH Server
+    description: >-
+      secure ssh server configuration
+    status: automated
+    rules:
+      - file_groupowner_sshd_config
+      - file_owner_sshd_config
+      - file_permissions_sshd_config
+      - file_permissions_sshd_private_key
+      - file_permissions_sshd_pub_key
+      - sshd_set_loglevel_verbose
+      - sshd_enable_pam
+      - sshd_disable_root_login
+      - disable_host_auth
+      - sshd_disable_empty_passwords
+      - sshd_do_not_permit_user_env
+      - sshd_disable_rhosts
+      - sshd_disable_x11_forwarding
+      - sshd_disable_tcp_forwarding
+      - sshd_max_auth_tries_value=4
+      - sshd_set_max_auth_tries
+      - sshd_set_maxstartups
+      - var_sshd_set_maxstartups=10:30:60
+      - sshd_set_max_sessions
+      - var_sshd_max_sessions=10
+      - sshd_set_login_grace_time
+      - var_sshd_set_login_grace_time=60
+      - sshd_idle_timeout_value=15_minutes
+      - sshd_set_idle_timeout
+      - sshd_set_keepalive
+      - var_sshd_set_keepalive=0
+      - sshd_x11_use_localhost
+      - sshd_disable_kerb_auth
+      - sshd_disable_gssapi_auth
+      - sshd_enable_strictmodes
+      - sshd_rekey_limit
+      - var_rekey_limit_size=1G
+      - var_rekey_limit_time=1hour
+      - sshd_use_strong_rng
+      - sshd_set_keepalive_0
+  ###
+  ### 5. Networking
+  ###
+  - id: 5.1
+    title: General network configuration
+    description: >-
+      If the user did not configure IPv6 on the system and it is not needed, it should be disabled.
+    status: partial
+    rules:
+      - kernel_module_sctp_disabled
+      - kernel_module_dccp_disabled
+      - sysctl_net_ipv4_ip_forward
+      - sysctl_net_ipv6_conf_all_forwarding
```

Comment:
Is the use of VMs a supported use case for the profiles? 

Once IP forwarding is disabled in a host the guest VMs will loose access to the external world.

Reply:
Thanks for pointing this out, I suppose this profile should be VM-friendly as well. I looked at the two rules, and the ipv4 one has a jinja macro in the platform: https://github.com/ComplianceAsCode/content/blob/f31f13744364d29489e2a414a6a3fd43957ca3bd/shared/macros/01-general.jinja#LL972C1-L977C17 that disables the rule if rhel8 product uses oVirt. I could create a new macro that would apply the machine platform for fedora product only, but I wonder if that could have adverse effect on other potential fedora profiles that do not consider the VM usecase. What are your thoughts on such a solution @yuumasato?

Reply:
You may be able to use CPE applicability language here. Something like `package[libvirtd]` as a platform on Fedora might do it.

Reply:
I guess making the rules for IP forwarding not applicable when an VM hypervisor is installed makes sense.

But could there be usecases that expect VMs to not have access to the external world?

Reply:
The problem is that libvirt is installed by default on Fedora 38 Workstation:
![image](https://user-images.githubusercontent.com/87149490/236617931-5e55c214-eb75-49fc-af67-b27d6bd70e05.png)
The purpose of this profile is to increase security but at the same time maintain usability, so I believe that users should be able to run VMs on their machines without connectivity problems (most users probably want networking inside their VMs). I am not sure there is a way to include these rules in the profile.